lib/tests/tap_syntax.rb

```diff
@@ -3,10 +3,18 @@
 module Homebrew
   module Tests
     class TapSyntax < Test
+      # Eventually these should be all `tap.official?` taps except core & cask?
```

Comment:
can we filter some things out of that list instead of hardcoding the preferred list?

Reply:
I actually think the filtering should instead be based on tap contents e.g. a `git grep -qE '^# typed: (strict|true|strong)$'` (which will exit immediately on finding the first result and is very fast)

Reply:
Could also/instead look for the presence of any `.rbi` files in `git ls-files`.

Reply:
I suggest a `tap.sorbet_support?` method in `brew` that checks for Sorbet support in the tap. This can probably also be used by `brew typecheck` to error out if a tap doesn't support typechecking.

Reply:
All of those are much better ideas! Thanks! I'll do that.

Reply:
Eventually, yeh, but for now I think it makes sense to just keep this scoped to `brew test-bot` until we decide whether we want to consider supporting this for third-party taps.

Reply:
I'm confused (sorry, too much in my brain), should I:

- Make the changes to add `tap.sorbet_supported?` now.
- Not make the changes, ship this as is.
- Not make the changes, close this PR and (for now) add a `brew typecheck homebrew/tap` CI step to the individual taps where I've added Sorbet, making the changes to `tap.sorbet_supported?` later.
- Something else.

Reply:
> * Something else.

@issyl0 Sorry for the confusion! Something else but: I've made the suggested code changes here. They need to be tested but I think they should land before adding `brew typecheck` to any other workflows and, when this is merged, the other repos where `brew typecheck` was added to workflows that already run `brew test-bot --tap-syntax` can have those steps removed.

Makes sense?

Reply:
Makes sense, yep!